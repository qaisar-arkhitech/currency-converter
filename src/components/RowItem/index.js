import React, {useContext} from "react"
import {Text, TouchableOpacity, View} from "react-native"
import PropTypes from "prop-types"
import themeStyles from "./styles"
import {ThemeContext} from "../../ContextUtils/ThemeContext"

// RowItem Component content
export const RowItem = ({title, onPress, rightIcon, ...rest}) => {
  const {styleableTheme} = useContext(ThemeContext)
  const styles = themeStyles(styleableTheme)
  return (
    <TouchableOpacity onPress={onPress} style={styles.row} {...rest}>
      <Text style={styles.title}>{title}</Text>
      {rightIcon}
    </TouchableOpacity>
  )
}

// RowItem Proptypes
RowItem.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  rightIcon: PropTypes.node,
}

// RowItem Default props
RowItem.defaultProps = {
  onPress: () => {},
  rightIcon: null,
}

export const RowSeparator = ({style}) => {
  const {styleableTheme} = useContext(ThemeContext)
  const styles = themeStyles(styleableTheme)
  return <View style={[styles.separator, style]} />
}

// RowSeparator Proptypes
RowSeparator.propTypes = {
  style: PropTypes.object.isRequired,
}

// RowSeparator Default props
RowSeparator.defaultProps = {}
