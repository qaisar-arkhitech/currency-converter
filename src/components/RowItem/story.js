import React from "react"
import RowItem from "."

export default {
  title: "RowItem",
  storyFn() {
    return (
      <div>
        <RowItem />
      </div>
    )
  },
}
