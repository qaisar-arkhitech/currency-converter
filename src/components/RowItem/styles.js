import {StyleSheet} from "react-native"

// RowItem: Styling here
const styles = StyleSheet.create({
  row: {
    paddingHorizontal: 20,
    paddingVertical: 16,
    justifyContent: "space-between",
    alignItems: "center",
    flexDirection: "row",
    backgroundColor: "#fff",
  },
  title: {
    fontSize: 16,
  },
  separator: {
    height: StyleSheet.hairlineWidth,
  },
})

export default styles
