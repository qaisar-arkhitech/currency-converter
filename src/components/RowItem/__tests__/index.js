import React from "react"
import renderer from "react-test-renderer"
import RowItem from ".."

const createProps = () => ({
  classes: {},
})

describe("<RowItem />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<RowItem {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
