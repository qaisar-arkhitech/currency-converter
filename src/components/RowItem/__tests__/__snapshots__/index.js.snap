// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowItem /> Does not explode 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  />
</View>
`;
